### PR TITLE
Fixed some api changes in Lua5.3: http://www.lua.org/manual/5.3/manua…

### PR DESCRIPTION
**Pull request by gczuczy**
For Issue74 https://github.com/jeremyong/Selene/issues/74

Should fix these: http://www.lua.org/manual/5.3/manual.html#8.3
Please review, and use it if you think it's good enough.

### CODE REVIEW

**Comment by jeremyong** [comment]
Looks good but no hard tabs please :)

**Comment by gczuczy** [comment]
Thanks. Do you need me to fix the tabs, or could you just please adjust it? Sorry for missing your format guide, I just wanted to make it work :)

**Comment by jeremyong** [comment]
Not a problem. Feel free to fix it on your end whenever you have time. Thanks!

**Comment by gczuczy** [comment]
It's untabified now, please check.

File: include/selene/primitives.h [hunk 1]
```diff
@@ -94,11 +94,17 @@ inline T _check_get(_id<T&&>, lua_State *l, const int index) {
 
 
 inline int _check_get(_id<int>, lua_State *l, const int index) {
+#if LUA_VERSION_NUM >= 503
+    return luaL_checkinteger(l, index);
+#else
     return luaL_checkint(l, index);
```
Comment (daurnimator) [inline include/selene/primitives.h]: should just use `luaL_checkinteger` in all cases. `luaL_checkint` is only a cast.

File: include/selene/primitives.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ inline int _get(_id<int>, lua_State *l, const int index) {
 }
 
 inline unsigned int _get(_id<unsigned int>, lua_State *l, const int index) {
-#if LUA_VERSION_NUM >= 502
+#if LUA_VERSION_NUM >= 502 && LUA_VERSION_NUM < 503
```
Comment (daurnimator) [inline include/selene/primitives.h]: should just be `==`

File: include/selene/primitives.h (hunk 1 above)
Comment (gczuczy) [inline include/selene/primitives.h]: The pull request is closed, so I guess you are free to submit one yourself with these fixes?

**Pull request merged**